net/portmapper/portmapper.go

```diff
@@ -835,3 +857,45 @@ var uPnPPacket = []byte("M-SEARCH * HTTP/1.1\r\n" +
 	"ST: ssdp:all\r\n" +
 	"MAN: \"ssdp:discover\"\r\n" +
 	"MX: 2\r\n\r\n")
+
+var (
```

Comment:
this does PMP and PCP... want to do UPnP while you're here?

also, send a metric for each about whether we send a discovery/probe request at all? would be useful for battery debugging.

Reply:
ok. next week, i'm out of typing for today.

Reply:
Ping, if you forgot this.

Reply:
Thanks for the reminder. Now with UPnP metrics, a few more PXP metrics, and a bonus minor fix.